### ISSUE TITLE
Fix misaligned pointer dereference on Rust beta and nightly

### ISSUE THREAD

**Issue by PierreZ** [foundationdb-rs/foundationdb-rs#90]
[CI link](https://github.com/foundationdb-rs/foundationdb-rs/actions/runs/4755251848/jobs/8449150843)
```
thread 'test_range' panicked at 'misaligned pointer dereference: address must be a multiple of 0x4 but is 0x7f0e5986d166', foundationdb/src/future.rs:308:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'test_range' panicked at 'panic in a function that cannot unwind', library/core/src/panicking.rs:126:5

```

**Comment by behos** [comment]
Aligning the pointers is possible (see #91 ) but the failure is caused by a new debug assertion:
https://github.com/rust-lang/rust/pull/98112

A different workaround could be to disable debug assertions in the CI through CARGO_PROFILE_DEV_DEBUG_ASSERTIONS=false

**Comment by PierreZ** [comment]
Fixed by #95
